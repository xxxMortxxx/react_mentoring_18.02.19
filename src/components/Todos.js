import React from 'react';

export default class Todos extends React.Component {
  state = {
    todos: [],
    inputText: ''
  }

  onAddButtonClick = () => {
    this.setState(({todos, inputText}) => ({
      todos: [...todos, inputText],
      inputText: ''
    }))
  }

  render() {
    return (
      <div>
        <input type="text"
               value={this.state.inputText}
               onChange={({target: {value}}) => this.setState({inputText: value})}
        />
        <button onClick={this.onAddButtonClick}>Add</button>
        <ul>{
          this.state.todos.map((todo, index) => {
            return (
              <li key={index}>{todo}</li>
            )
          })
        }</ul>
      </div>

    )
  }

}