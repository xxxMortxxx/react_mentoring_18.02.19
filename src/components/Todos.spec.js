import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Todos from "components/Todos";

describe('Todos', () => {
 it('should render as expected', () => {
   const tree = renderer.create(<Todos/>).toJSON();

   expect(tree).toMatchSnapshot();
 });

 it('should add new todo', () => {
   const wrapper = shallow(<Todos/>);

   expect(wrapper.find('li').length).toBe(0);

   wrapper
     .find('input')
     .simulate('change', {target: {value: 'new todo'}});

   wrapper.find('button').simulate('click');

   expect(wrapper.find('li').text()).toBe('new todo');
 })
});