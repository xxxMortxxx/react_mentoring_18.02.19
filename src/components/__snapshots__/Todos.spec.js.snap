// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos should render as expected 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
  <ul />
</div>
`;
